idea_town/frontend/static-src/app/lib/page-manager.js

```diff
@@ -37,6 +41,16 @@ export default class PageManager {
       const Ctor = this.pages[page];
       const newPage = new Ctor(opts);
 
+      if (newPage.headerScroll) {
```

Comment:
@meandavejustice About your gist: I think it might be worth splitting out a base page view (has a header, is managed by the page manager) from the base view (other reusable stuff that applies to non-page level modules).